src/frontend/src/optimizer/plan_node/predicate_pushdown.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use paste::paste;
+
+use super::*;
+use crate::utils::Condition;
+use crate::{for_batch_plan_nodes, for_stream_plan_nodes};
+/// The trait for predicate pushdown, only logical plan node will use it, though all plan node impl
+/// it.
+pub trait PredicatePushdown {
+    /// Push predicate down for every logical plan node.
+    ///
+    /// If you don't know how to implement this trait, you can just use [`gen_new_node`] or
+    /// [`gen_filter_above`] depending on whether the plan node has ONLY one input or not.
```

Comment:
Please describe the cases more accurately (maybe copy part of the descriptions in RFC), and remove "If you don't know how to implement". Use precise and concise words.